### PR TITLE
bugfix: used 'remove_tree' to remove directories when TEST_NGINX_USE_…

### PR DESCRIPTION
**Pull request by spacewander**
…HUP is specified for consistency.

When TEST_NGINX_USE_HUP is not specified, Test-Nginx will use
'remove_tree' to remove undesired directories. We need to do
the same when TEST_NGINX_USE_HUP is specified for consistency.

### CODE REVIEW

File: lib/Test/Nginx/Util.pm [hunk 1]
```diff
@@ -728,32 +728,25 @@ sub run_tests () {
 
 sub setup_server_root () {
     if (-d $ServRoot) {
+        # Take special care, so we won't accidentally remove
+        # real user data when TEST_NGINX_SERVROOT is mis-used.
+        remove_tree($ConfDir, $HtmlDir, $LogDir, glob("$ServRoot/*_cache"),
+                    glob("$ServRoot/*_temp"));
```
Comment (agentzh) [inline lib/Test/Nginx/Util.pm]: This does not look right. You already remove the nginx.pid file which we want to keep in the HUP reload mode.

Comment (spacewander) [reply lib/Test/Nginx/Util.pm]: @agentzh 
Fixed in the new commit.

**Comment by agentzh** [comment]
@spacewander What exact problems are you trying to solve here? Better make it explicit in your commit log message. Simply saying "we should do something" without saying what problems to solve is too vague.

**Comment by spacewander** [comment]
@agentzh 
Just update the commit message to show the problem this PR solved.

**Pull request merged**